src/AutoRest.CSharp.V3/Output/Models/Types/ObjectType.cs

```diff
@@ -212,10 +217,18 @@ private ObjectTypeConstructor BuildInitializationConstructor()
                         defaultParameterValue = BuilderHelpers.ParseConstant(defaultValueObject, propertyType);
                     }
 
+                    defaultInitializationValue = defaultParameterValue;
```

Comment:
Should this variable be set inside the if?

Reply:
It doesn't matter, it can be inside or outside the result would be the same.